You can now tape people's mouths closed (Also tape GAGS)

Ebb-Real opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
With the new kidnapping objectives and tape currently only having one use I made this PR. You can now break off pieces of tape from tape rolls and use them as a muzzle. Different tape types have different effects like pointy tape dealing damage to you if its pulled off by someone else than yourself. Or super sticky tape taking longer to pull off in the stripping menu.

Tape rolls are also GAGS now so more types of tape can easily be added without having to make new sprites if someone wanted to. Super tape is now black instead of gold because its hard to have one shaded color look like actual gold and I thought black tape would make sense since strong tape usually is colored black. Regular sticky tape now is grey like duct tape and surgical tape is still silver colored. All the darker colored tapes are super tape.

Super pointy tape is now actually obtainable in the game now with illegal tech and traitor uplinks.

Tape now makes sounds.

Now we can finally use GAGS to gag people!

![Tape Gags](https://user-images.githubusercontent.com/95765134/173194411-4e2a8557-138f-4e19-be27-3948782f2cb3.png)

Tape sounds are from the free sound library: https://www.fesliyanstudios.com/royalty-free-sound-effects-download/duct-tape-66

## Why It's Good For The Game

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->
Tape now have more uses like the all purpose item it is, and super pointy tape is now obtainable through illegal tech and the traitor uplink since the syndicate enjoy torture tape. There's more fun options for kidnapping.

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl: Ebb
add: You can now rip pieces of tape off of tape rolls and use them to tape people's mouth closed by putting it on them as a mask. Either through the stripping menu or hitting them with the tape while not on combat mode. Different tape piece types have different properties.
add: Super pointy tape is now obtainable in-game by researching illegal tech.
add: Super pointy tape is now available in the traitor uplink.
soundadd: Tape now makes sounds.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Ebb-Real committed:
tape

Colovorat commented:
I'm just amazed that it's not a feature already in 2022

Fikou commented on code/game/objects/items/stacks/tape.dm:
```diff
@@ -15,6 +15,23 @@
+	if(user.get_inactive_held_item() == src)
+		if(is_zero_amount(delete_if_zero = TRUE))
+			return
+		user.put_in_hands(new tape_gag)
```
what if it fails

Ebb-Real replied:
did melbert's commit fix what you were talking about?

Ebb-Real replied:
I changed it to work similarly to how bluespace polycrystals do it. Should work with `put_in_hands`'s built system to drop the item on the floor if it fails.

Ebb-Real committed:
returns

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

Ebb-Real committed:
attack()

Ebb-Real commented on code/modules/clothing/masks/muzzle.dm:
```diff
@@ -25,3 +25,52 @@
 	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 100, FIRE = 0, ACID = 0)
 	equip_delay_other = 25 // my sprite has 4 straps, a-la a head harness. takes a while to equip, longer than a muzzle
+
+/obj/item/clothing/mask/muzzle/tape
```
I couldn't figure out a way to get the attacker's targeted body zone since `target_zone` didn't work but I thought it would be fine to just have it be any target since you won't use it on people for any other reason, like cuffs.

Ebb-Real committed:
unnecessary return

Ebb-Real committed:
test

Ebb-Real committed:
test2

Ebb-Real committed:
test 3

Ebb-Real committed:
WHERE IS IT

Ebb-Real committed:
just in case it actually makes a difference

Ebb-Real commented:
Wow I didn't expect that to actually work.

Ebb-Real committed:
dots

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

Ebb-Real commented on code/modules/clothing/masks/muzzle.dm:
```diff
@@ -25,3 +25,62 @@
+	greyscale_colors = "#B2B2B2"
+	moth_snack = null
+
+/obj/item/clothing/mask/muzzle/tape/attack(mob/living/M, mob/living/user, params)
```
`NOBLUDGEON` seems to stop the attack entirely.

Ebb-Real replied:
I think I just did it wrong the whole time.

Ebb-Real committed:
var names and mouth covered check

Ebb-Real committed:
targeted taping

Ebb-Real commented on code/modules/clothing/masks/muzzle.dm:
```diff
@@ -25,3 +25,62 @@
+	greyscale_config = /datum/greyscale_config/tape_piece
+	greyscale_config_worn = /datum/greyscale_config/tape_piece/worn
+	greyscale_colors = "#B2B2B2"
+	moth_snack = null
```
Can you elaborate on this @MrMelbert ? And what should I do to improve it?

Ebb-Real replied:
What im trying to do with it is stop it from being a moth snack so it dosen't conflict with its `/attack` proc just for clarification.

Wallemations replied:
How do other inedible items use it? Search the variable up and give it a check

Ebb-Real replied:
I did but the only things that show up are in relation to making the var work and the code from this PR.
![image](https://user-images.githubusercontent.com/95765134/174482364-6e69d14d-584a-4ffb-a906-2ba136dddf83.png)

I don't think there are any inedible clothing to moths in the code already.

Ebb-Real replied:
I think what melbert was talking about is this i just found...
![image](https://user-images.githubusercontent.com/95765134/174482526-078ca512-954a-4a4a-ae1e-e974529983dc.png)
But how could you stop this for specific items

Ebb-Real replied:
Setting it to null seems to work in playtest so thats kinda weird

Ebb-Real committed:
cleanup

Miczu555PL commented:
How about also adding tape to traitor uplink?

Ebb-Real commented:
That would probably be too powerful since the super pointy tape adds 100% embed chance to items. Meaning every traitor could buy a bunch of it roundstart and make 100% embed spears every round.

Likteer commented:
> That would probably be too powerful since the super pointy tape adds 100% embed chance to items. Meaning every traitor could buy a bunch of it roundstart and make 100% embed spears and grenades every round.

Isn't stuff locked behind BEPIS ridiculous in this time and age? Sticky grenades can fuck the user over with RNG and it's not that bad for 100% spears if it's from a traitor uplink.

Ebb-Real commented:
> > That would probably be too powerful since the super pointy tape adds 100% embed chance to items. Meaning every traitor could buy a bunch of it roundstart and make 100% embed spears and grenades every round.
> 
> Isn't stuff locked behind BEPIS ridiculous in this time and age? Sticky grenades can fuck the user over with RNG and it's not that bad for 100% spears if it's from a traitor uplink.

100% embed spears deal both high throw damage and embedding damage. This damage is enough to crit in 3 hits (The same as a 357. revolver) You can have 2 spears in your hands and drag one behind which are taped with it. You almost immediately after roundstart and have a budget 357 that kill crit anyone with 3 throws.

As for grenades you can put 100u potassium and 100u water in a grenade and tape it to have a 100% embed chance and it will instakill anyone you throw it into with pretty much no counter. 

Traitors having all this roundstart and with almost no effort makes for a very boring and unbalanced traitor meta.

Ebb-Real committed:
I accidentally broke tape_gag

ATH1909 commented:
> As for grenades you can put 100u potassium and 100u water in a grenade and tape it to have a 100% embed chance and it will instakill anyone you throw it into with pretty much no counter.

```
/obj/item/grenade/proc/botch_check(mob/living/carbon/human/user)
	if(sticky && prob(50)) // to add risk to sticky tape grenade cheese, no return cause we still prime as normal after.
		to_chat(user, span_warning("What the... [src] is stuck to your hand!"))
		ADD_TRAIT(src, TRAIT_NODROP, STICKY_NODROP)
```

'course, I think you can still get around this by just sticky taping an evidence bag or something, but I figured I should mention that sticky 'nading is not *quite* as easy/risk-free as you think it is

Likteer commented:
@Ebb-Real I've mentioned RNG fucking you over with grenades, ATH just elaborated on that. 
Syndicate revolver also crits people in 2 hits, not 3. Even if the 3 spears collectively can bring someone to crit - reaction time between throws for both sides and the obviousness of the sprites - it really isn't cmparable to a gun. Plus they can be caught and have slower projectile speed (?).
Embedding tape really seems like a great idea for the traitor uplink, I urge you to reconsider.

Ebb-Real commented:
I forgot about the taping to hand. Tested it with evidence bag and its still 

> 

Forgot this existed. Also tested it with evidence bag and its impossible as far as i know, since when you ignite the grenade it will stick to your hand and be unable to move.

As for spears you most likely won't get them back after using them so even if its used fast it can only be done once. So its probably fine to add it.

Ebb-Real committed:
uplink

Ebb-Real committed:
tips and grammer

Ebb-Real committed:
Merge branch 'master'

Conflicts:
	code/modules/research/techweb/all_nodes.dm

Ebb-Real committed:
i dunno how i had this before merge but its cool

Ebb-Real commented:
wonder what caused that failed check

Ebb-Real committed:
test

Ebb-Real committed:
it was just bots being weird?

Ebb-Real committed:
inedible flag

Ebb-Real commented:
First time making new flags so tell me if im making a mistake with it.

Works in playtest.

Ebb-Real committed:
how did i type it like this

JohnFulpWillard commented on code/modules/clothing/masks/muzzle.dm:
```diff
@@ -25,3 +25,74 @@
+	name = "super pointy tape piece"
+	desc = "A piece of tape that can be put over someone's mouth. This thing could rip your face into a thousand pieces if ripped off carelessly."
+	greyscale_colors = "#8C0A00#300008"
+	strip_delay = 60
```
intentional that this is faster than regular super tape?

Ebb-Real replied:
Yes, so there's a slight reason to use super sticky tape over super pointy other than it being non-lethal. I also like the idea of using it in a kidnapping situation to make people speak by ripping it off repeatedly.

JohnFulpWillard commented on code/__DEFINES/obj_flags.dm:
```diff
@@ -66,6 +66,8 @@
 /// Usable as casting clothes by wizards (only matters for suits and headwear)
 #define CASTING_CLOTHES (1<<19)
+///Moths can't eat the clothing that has this flag.
+#define INEDIBLE (1<<20)
```
could this use something more descriptive? INEDIBLE_CLOTHING or something.

Ebb-Real committed:
INEDIBLE_CLOTHING

Ebb-Real committed:
what

Ebb-Real committed:
non-combat mode applying

Ebb-Real commented:
thought applying the tape was a little clunky with the targeting so i changed it to work if the attacker isn't on combat mode. Removed `NOBLUDGEON` to make it clear that the attacker is hitting them with it instead of applying if on combat mode.

Ebb-Real committed:
damage on ripping off by yourself too

Ebb-Real committed:
tape sounds

Ebb-Real committed:
a pain

MrDoomBringer commented:
Hate to suggest this while the PR has been open for so long, but I'd be super chuffed if you replaced the normal tape color with White, and the surgical tape color with a light blue. 
- The white tape roll looks much more like scotch tape than something medical
- A blue tape roll will look like it's for a special purpose, and would fit medical's theme

Ebb-Real committed:
blue surgical tape

Ebb-Real commented:
This is how it looks now:
![image](https://user-images.githubusercontent.com/95765134/178545951-260a56bb-c9d3-493f-8b1e-ae5becaa2d4f.png)

Ebb-Real committed:
lighter blue

Ebb-Real committed:
i don't know why this was here in the first place

JohnFulpWillard commented on code/modules/clothing/masks/muzzle.dm:
```diff
@@ -25,3 +25,79 @@
+	greyscale_config = /datum/greyscale_config/tape_piece
+	greyscale_config_worn = /datum/greyscale_config/tape_piece/worn
+	greyscale_colors = "#B2B2B2"
+	var/harmful_strip = FALSE
+	var/stripping_damage = 0
```
can you add autodoc comments please

JohnFulpWillard reviewed (commented):
small comment then it's fine

Ebb-Real committed:
Autodocs

Ebb-Real committed:
h
